### PR TITLE
Log Operation name

### PR DESCRIPTION
**Pull request by kamilszostak**
* Log operation name with every track request - #94 
* Fix absolute URL encoding bug

### CODE REVIEW

File: AutoCollection/RequestDataHelper.ts [hunk 1]
```diff
@@ -96,11 +98,14 @@ class RequestDataHelper {
         }
         
         var encrypted = <any>request.connection ? (<any>request.connection).encrypted : null;
+        var pathName = request.url.split('?')[0];
+        var search = request.url.indexOf('?') != -1 ? request.url.substring(request.url.indexOf('?')) : null;
```
Comment (dnduffy) [inline AutoCollection/RequestDataHelper.ts]: Aren't these kinds of properties already available directly?

**Comment by kartang** [comment]
LGTM

File: Tests/AutoCollection/RequestDataHelper.tests.ts [hunk 2]
```diff
@@ -33,15 +33,49 @@ describe("AutoCollection/RequestDataHelper", () => {
...
         });
 		
 		it("should return an absolute url for encrypted traffic", () => {
 			request.connection.encrypted = true;
 			
 			var helper = new RequestDataHelper(<any>request);
 			var requestData = helper.getRequestData();
-			assert.equal(requestData.baseData.url, "https://bing.com/search%3Fq=test");
+			assert.equal(requestData.baseData.url, "https://bing.com/search?q=test");
+        });
```
Comment (kartang) [inline Tests/AutoCollection/RequestDataHelper.tests.ts]: May be add tests for some border case scenarios like when the actual path name is empty, operation name might have to be "GET /" as opposed to "GET ".

File: Tests/AutoCollection/RequestDataHelper.tests.ts [hunk 3]
```diff
@@ -33,15 +33,49 @@ describe("AutoCollection/RequestDataHelper", () => {
...
+            url: "/a/b/c/?q=test&test2",
+            connection: {
+                encrypted: false
+            },
+            headers: {
+                host: "bing.com"
+            }
+		}
+		
+		it("should return an absolute url for  complext urls", () => {
```
Comment (dnduffy) [inline Tests/AutoCollection/RequestDataHelper.tests.ts]: extra spaces in "for  complext" and an extra "t" on "complex".

File: Tests/Library/Client.tests.ts [hunk 4]
```diff
@@ -247,6 +247,23 @@ describe("Library/Client", () => {
...
+                client.trackRequest(<any>request, <any>response, properties);
+                
+                // emit finish event
+                response.emitFinish();
+                
+                // validate
+                var args = trackStub.args;
+                var tags = args[0][1];
+                
+                assert.equal(tags["ai.operation.name"], "method /search");
```
Comment (dnduffy) [inline Tests/Library/Client.tests.ts]: Why is the method "method" and not "GET" or "POST"?

File: Tests/Library/Client.tests.ts [hunk 5]
```diff
@@ -258,6 +258,8 @@ describe("Library/Client", () => {
                 // validate
                 var args = trackStub.args;
                 var tags = args[0][1];
+
+                assert.equal(tags["ai.operation.name"], "method /search");
```
Comment (dnduffy) [inline Tests/Library/Client.tests.ts]: Why is the method "method" and not "GET" or "POST"?

**Comment by dnduffy** [comment]
:shipit:

File: Tests/Library/Client.tests.ts (hunk 5 above)
Comment (kamilszostak) [inline Tests/Library/Client.tests.ts]: It's just a sample value - See line 180. I changed it to "GET" and updated the unit test.

File: Tests/AutoCollection/RequestDataHelper.tests.ts (hunk 3 above)
Comment (kamilszostak) [inline Tests/AutoCollection/RequestDataHelper.tests.ts]: Fixed.

File: Tests/AutoCollection/RequestDataHelper.tests.ts (hunk 2 above)
Comment (kamilszostak) [inline Tests/AutoCollection/RequestDataHelper.tests.ts]: I added unit tests to cover corner cases.

File: AutoCollection/RequestDataHelper.ts (hunk 1 above)
Comment (kamilszostak) [inline AutoCollection/RequestDataHelper.ts]: I changed the code to use the URL lib. Thx!
